### PR TITLE
[1LP][RFR]Automated Service state test

### PR DESCRIPTION
**Pull request by niyazRedhat**
### PRT Run
{{pytest: cfme/tests/services/test_myservice.py::test_service_state --use-provider vsphere67-nested --long-running }}

### CODE REVIEW

File: cfme/tests/services/test_myservice.py [hunk 1]
```diff
@@ -208,7 +210,27 @@ def test_service_state():
             3.
             4. Service State should be Provisioned or Failed
     """
-    pass
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    service = MyService(appliance, catalog_item.name)
+    service_request = service_catalogs.order()
```
Comment (sshveta) [inline cfme/tests/services/test_myservice.py]: You can use fixture 'order_service' that will order catalog_item and also delete service at the end.

https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/service_fixtures.py#L161

File: cfme/tests/services/test_myservice.py [hunk 2]
```diff
@@ -184,11 +186,11 @@ def test_retire_on_date_for_multiple_service():
     pass
 
 
-@pytest.mark.meta(coverage=[1678123])
-@pytest.mark.manual
+@pytest.mark.meta(automates=[1678123])
 @pytest.mark.ignore_stream('5.10')
 @pytest.mark.tier(2)
-def test_service_state():
+@pytest.mark.parametrize("check", ["provisioned", "un_provisioned"])
```
Comment (kedark3) [inline cfme/tests/services/test_myservice.py]: Minor suggestion: please replace `un_provisioned` <==> `unprovisioned`

Comment (john-dupuy) [reply cfme/tests/services/test_myservice.py]: Agree, I'd say it's _optional_ though

**Comment by john-dupuy** [comment]
Please check PRT for 5.10

File: cfme/tests/services/test_myservice.py (hunk 1 above)
Comment (niyazRedhat) [reply cfme/tests/services/test_myservice.py]: @sshveta  we need to fail ordered service catalog item before successfully provisioned. so for that, we are removing provider while ordered catalog item is in progress hence we can not use fixture which provides successfully provision service catalog item.

File: cfme/tests/services/test_myservice.py [hunk 3]
```diff
@@ -208,7 +210,27 @@ def test_service_state():
...
+        provider.delete_rest()
+        provider.wait_for_delete()
+
+    view = navigate_to(service, "Details")
+    wait_for(lambda: view.entities.lifecycle.get_text_of("State") == expected_state,
+             fail_condition=0,
+             num_sec=300,
+             fail_func=view.browser.refresh)
+
+    @request.addfinalizer
```
Comment (digitronik) [inline cfme/tests/services/test_myservice.py]: Always better to add finalizer where we create entity.

**Pull request merged**